kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -318,11 +319,11 @@ func NewDispatcher(args *KafkaDispatcherArgs) (*KafkaDispatcher, error) {
 	return dispatcher, nil
 }
 
-func (d *KafkaDispatcher) getChannelReferenceFromHost(host string) (eventingchannels.ChannelReference, error) {
+func (d *KafkaDispatcher) getChannelReferenceFromHost(host url.URL) (eventingchannels.ChannelReference, error) {
 	chMap := d.getHostToChannelMap()
-	cr, ok := chMap[host]
+	cr, ok := chMap[host.Path]
 	if !ok {
-		return cr, fmt.Errorf("invalid Hostname:%s. Hostname not found in ConfigMap for any Channel", host)
+		return cr, fmt.Errorf("invalid Hostname:%v. Hostname not found in ConfigMap for any Channel", host)
```

Comment:
(not related to your changes) I doubt the absence of space between `:` and the string was intentional. Shall we use the occasion to fix the format?

Reply:
+1

Reply:
Done.